### PR TITLE
Add nullability to antiforgery

### PR DESCRIPTION
**Pull request by pranavkm**
Addresses #5680

### CODE REVIEW

File: src/Antiforgery/test/DefaultAntiforgeryTokenGeneratorTest.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+#nullable disable
```
Comment (pranavkm) [inline src/Antiforgery/test/DefaultAntiforgeryTokenGeneratorTest.cs:4]: This one's a massive pain. I'm leaving this as an exercise for the future.

Comment (JamesNK) [reply src/Antiforgery/test/DefaultAntiforgeryTokenGeneratorTest.cs:4]: I didn't enable null in Newtonsoft.Json tests because of the extra work involved. A couple of APIs in Newtonsoft.Json were marked as not null that otherwise should have allowed null. Null in tests would have caught those. However I think these kind of issues should be caught in previews or otherwise corrected in servicing.

tldr, nullable in tests is nice, but I don't think it is a requirement when difficult

File: src/Antiforgery/test/DefaultAntiforgeryTokenGeneratorTest.cs [hunk 2]
```diff
@@ -621,3 +622,4 @@ public override string Name
         }
     }
 }
+#nullable restore
```
Comment (JamesNK) [inline src/Antiforgery/test/DefaultAntiforgeryTokenGeneratorTest.cs:625]: I don't think this is required

**Comment by pranavkm** [comment]
@javiercn would you mind looking at this? Most of this is fairly mechanical.

File: src/Antiforgery/src/AntiforgeryTokenSet.cs [hunk 3]
```diff
@@ -18,10 +18,10 @@ public class AntiforgeryTokenSet
         /// <param name="formFieldName">The name of the form field used for the request token.</param>
         /// <param name="headerName">The name of the header used for the request token.</param>
         public AntiforgeryTokenSet(
-            string requestToken,
-            string cookieToken,
+            string? requestToken,
+            string? cookieToken,
```
Comment (javiercn) [inline src/Antiforgery/src/AntiforgeryTokenSet.cs:22]: Is this correct? I would expect these fields to be non-nullable. What's the use case for creating a token set with all null values. Also, why is the formfield non-nullable (apart from the check we do in the ctor)

Comment (pranavkm) [reply src/Antiforgery/src/AntiforgeryTokenSet.cs:22]: I've been trying to walk backwards from the implementation. For instance, in this case, `formFieldName` was the only value that disallowed nullability.

Comment (javiercn) [reply src/Antiforgery/src/AntiforgeryTokenSet.cs:22]: I think we shouldn't be doing this mechanically and instead do what makes sense. I worry that if we just apply things mechanically we'll end up codifying behaviors that don't match the actual system expectations.

For example, it is very common that we avoid nullchecks in many cases but factually we still expect those types to not receive null values.

Comment (pranavkm) [reply src/Antiforgery/src/AntiforgeryTokenSet.cs:22]: I looked at this again, ad we anticipate these values to be null and have specific code paths for this: https://github.com/dotnet/aspnetcore/blob/master/src/Antiforgery/src/Internal/DefaultAntiforgery.cs#L115-L125

**Review by JamesNK — APPROVED** [review]
